extensions/script/lua/LuaBaseStream.cpp

```diff
@@ -54,15 +54,15 @@ std::string LuaBaseStream::read(size_t len) {
   // http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2012/n3337.pdf
   auto read = stream_->read(reinterpret_cast<uint8_t *>(&buffer[0]), static_cast<int>(len));
 
-  if (read != len) {
-    buffer.resize(static_cast<size_t >(read));
+  if (gsl::narrow<size_t>(read) != len) {
+    buffer.resize(gsl::narrow<size_t>(read));
```

Comment:
Unsafe narrowing: `read` can be `-1` in case of an error.

Reply:
Updated in 267cda83a77a5a6d05f9df714d0c923814351826